Idrall/Check.idr

```diff
@@ -376,6 +415,17 @@ mutual
        b' <- evalClosure bT (VNeutral aT (NVar x))
        b <- readBackTyped (extendCtx ctx x aT) (VConst CType) b'
        Right (EPi x a b)
+  readBackTyped ctx (VConst CType) (VHPi i ty f) =
+    let i' = freshen (ctxNames ctx) i in
+    do a <- readBackTyped ctx (VConst CType) ty
+       body' <- readBackTyped ctx (VConst CType) (f (VNeutral ty (NVar i')))
+       -- TODO not remotely sure about this ^^^, especially the CType
```

Comment:
The `CType` seems fine to me (not thinking too much)

Reply:
That's good enough for me!